### PR TITLE
MODULES-3958 Get rid of mocha-based stubbing across all modules

### PR DESCRIPTION
**Pull request by lionce**

### CODE REVIEW

File: spec/unit/provider/a2mod/gentoo_spec.rb [hunk 1]
```diff
@@ -32,9 +32,11 @@
     end
 
     it 'normalizes parameters' do
-      filetype.expects(:read).returns(%(APACHE2_OPTS="-D FOO -D BAR -D BAR"\n))
-      provider_class.expects(:filetype).returns(filetype)
-
+      # filetype.expects(:read).returns(%(APACHE2_OPTS="-D FOO -D BAR -D BAR"\n))
+      allow(filetype).to receive(:read).and_return(%(APACHE2_OPTS="-D FOO -D BAR -D BAR"\n))
```
Comment (david22swan) [inline spec/unit/provider/a2mod/gentoo_spec.rb]: This should be an expect.

File: spec/unit/provider/a2mod/gentoo_spec.rb [hunk 2]
```diff
@@ -32,9 +32,11 @@
...
+      # provider_class.expects(:filetype).returns(filetype)
+      allow(provider_class).to receive(:filetype)
```
Comment (david22swan) [inline spec/unit/provider/a2mod/gentoo_spec.rb]: Is there a reason for this?

File: spec/unit/provider/a2mod/gentoo_spec.rb (hunk 1 above)
Comment (ekohl) [reply spec/unit/provider/a2mod/gentoo_spec.rb]: Unless you add a `expect(filetype).to have_received(:read)` later on. IIRC there's a rubocop-rspec warning about this.

**Comment by david22swan** [comment]
Need's a clean up to remove commented code, but other than what I've mentioned it looks fine. Unsure about the removed unstubs, the code seem's to pass without them so it look's like the new stubs are overwriting the old one's.

**Pull request merged**